### ISSUE TITLE
better formatting of build-column error

### ISSUE THREAD

**Issue by shriram** [brownplt/pyret-lang#1584]
```
t = table: a, b
  row: 1, 2
  row: 3, 4
end

t.build-column("a", lam(r): 4 end)
```
The error comes out as

<img width="637" alt="image" src="https://user-images.githubusercontent.com/75030/116106591-3d55f400-a680-11eb-802d-9623e778f2a1.png">

which has a very non-built-in look, and furthermore, the parameter and the column names are written without strings, which could mislead some students.

**Comment by jpolitz** [comment]
Thanks, this we can definitely improve. Looks like a stringly-constructed message in a library rather than a custom error constructor.

**Comment by jpolitz** [comment]
Yeah, agreed. This is possible now by defining an object with render-reason and/or render-fancy-reason methods. But actually writing those methods in JS would be painful and needs a better API.
